### PR TITLE
eth_feeHistory

### PR DESCRIPTION
**Pull request by owenwahlgren**
# Description

Please provide a detailed description of what was done in this PR
Implementation of RPC method eth_feeHistory (evm-696)
# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Tested manually using param variations and bash. **Please note** no tests added in eth_endpoint_test.
Example request:
`curl http://localhost:10002/   -X POST   -H "Content-Type: application/json"   --data '{"method":"eth_feeHistory","params":[4, 5, [25, 75]],"id":1,"jsonrpc":"2.0"}'`

### CODE REVIEW

**Comment by owenwahlgren** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by vcastellm — COMMENTED** [review]
Can you check if `GetBlockByNumber` and other store operations have some kind of catching? If not, it would be great to add catching here, the operation is not that variable in time so a relatively short time caching will do and prevent the risk of DoS attacks.

File: scripts/cluster [hunk 1]
```diff
@@ -25,7 +25,7 @@ function initPolybftConsensus() {
 function createGenesis() {
   ./polygon-edge genesis $genesis_params \
     --block-gas-limit 10000000 \
-    --premine 0x85da99c8a7c2c95964c8efd687e95e632fc533d6:1000000000000000000000 \
+    --premine 0x1FC1411A3Bd09E63d4A306FD1EB838bA457ddA13:1000000000000000000000 \
```
Comment (vcastellm) [inline scripts/cluster]: Do not update this address please, it's used for load testing.

**Comment by goran-ethernal** [comment]
This is on par with `go-ethereum`. Really good job :100: .

I have couple of suggestions:
- Through https://github.com/0xPolygon/polygon-edge/pull/1629, we implemented `eth_maxPriorityFeePerGas`. In that PR we created `gasprice` package to contain these functions (`eth_maxPriorityFeePerGas, eth_feeHistory`, and others that we might introduce in future). My suggestion would be to move this `feeHistory` function to provided package and `GasHelper` struct, and expose that function to its interface. That interface is then used in `eth_endpoint.go`.
- It would be really awesome to have some unit tests for this. You can take a look at `gasprice_test.go` how we implemented them for `eth_maxPriorityFeePerGas`.

**Comment by owahlgren** [comment]
> This is on par with `go-ethereum`. Really good job 100 .
> 
> I have couple of suggestions:
> 
>     * Through [[EVM-697]: Implement `eth_maxPriorityFeePerGas` #1629](https://github.com/0xPolygon/polygon-edge/pull/1629), we implemented `eth_maxPriorityFeePerGas`. In that PR we created `gasprice` package to contain these functions (`eth_maxPriorityFeePerGas, eth_feeHistory`, and others that we might introduce in future). My suggestion would be to move this `feeHistory` function to provided package and `GasHelper` struct, and expose that function to its interface. That interface is then used in `eth_endpoint.go`.
> 
>     * It would be really awesome to have some unit tests for this. You can take a look at `gasprice_test.go` how we implemented them for `eth_maxPriorityFeePerGas`.

Thanks! Just saw this got merged 2 days ago. Will close this PR and reopen soon with functionality moved to the new `gasprice` package. Additionally I will do the checks mentioned by @vcastellm and provide some unit tests to `gasprice_test.go` :)

[EVM-697]: https://polygon.atlassian.net/browse/EVM-697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Pull request closed without merging**